roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -33,12 +33,13 @@
     IntType,
     check_get_frames_args,
     check_get_videos_args,
-    dict_recursive_update,
+    dict_deep_update,
 )
 from ...imagingextractor import ImagingExtractor
 from ...multisegmentationextractor import MultiSegmentationExtractor
 from ...segmentationextractor import SegmentationExtractor
 
+#from nwb_conversion_tools.utils.json_schema import dict_deep_update #debug
```

Comment:
@CodyCBakerPhD, curious if you have a tip here. Importing the `dict_deep_update` function from nwb_conversion_tools produces the error shown below. I gather the issue is that nwb_conversion_tools itself is dependent on roiextractors so the mutual dependency messes things up. For the current version of the PR, I have opted to copy the `dict_deep_update` function into extraction_tools within the roiextractors repo. Let me know if you have a more elegant idea.

```python
>>> import roiextractors
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/cesar/Repositories/roiextractors/roiextractors/__init__.py", line 3, in <module>
    from .extractorlist import *
  File "/Users/cesar/Repositories/roiextractors/roiextractors/extractorlist.py", line 7, in <module>
    from .extractors.nwbextractors import NwbImagingExtractor, NwbSegmentationExtractor
  File "/Users/cesar/Repositories/roiextractors/roiextractors/extractors/nwbextractors/__init__.py", line 1, in <module>
    from .nwbextractors import NwbImagingExtractor, NwbSegmentationExtractor
  File "/Users/cesar/Repositories/roiextractors/roiextractors/extractors/nwbextractors/nwbextractors.py", line 42, in <module>
    from nwb_conversion_tools.utils.json_schema import dict_deep_update #debug
  File "/Users/cesar/anaconda3/envs/nwb_conversion_env/lib/python3.7/site-packages/nwb_conversion_tools/__init__.py", line 3, in <module>
    from .datainterfaces import *
  File "/Users/cesar/anaconda3/envs/nwb_conversion_env/lib/python3.7/site-packages/nwb_conversion_tools/datainterfaces/__init__.py", line 35, in <module>
    from .ophys.caiman.caimandatainterface import CaimanSegmentationInterface
  File "/Users/cesar/anaconda3/envs/nwb_conversion_env/lib/python3.7/site-packages/nwb_conversion_tools/datainterfaces/ophys/caiman/caimandatainterface.py", line 1, in <module>
    from roiextractors import CaimanSegmentationExtractor
ImportError: cannot import name 'CaimanSegmentationExtractor' from 'roiextractors' (/Users/cesar/Repositories/roiextractors/roiextractors/__init__.py)
```

Reply:
Can you provide a minimal code snippet that reproduces this?

Like,
```
import roiextractors
import nwb_conversion_tools
```
with this branch of `roiextractors`?

Reply:
I was able to get it to work by doing

```
import nwb_conversion_tools
import roiextractors
```

but I do recognize this is not a good solution since someone might want to import just `roiextractors` independently. Really, the solution would be to propagate the JSON tools to a higher level (a dedicated repo), since this shows how generalized their usage is becoming.

However, that might be a long-term thought, and the short-term solution would really be to analyze if and why the deep update is so necessary for these local functions. I had thought that, in principle, the deep update is only something we really need over on `nwb_conversion_tools` for merging information across modalities/interfaces. Since `roiextractors`, as a module, builds its properties from the ground up there shouldn't be anything that really requires that level of merging detail.

Reply:
Just spoke to Ben about it; we'll go ahead and propogate the general JSON utils to a higher external repo so we can refer to it here in that fashion.

Reply:
> Just spoke to Ben about it; we'll go ahead and propogate the general JSON utils to a higher external repo so we can refer to it here in that fashion.

@CodyCBakerPhD  is this done?